### PR TITLE
Leecher to seeder

### PR DESCRIPTION
**Pull request by spersico**
This PR, when merged, will:
- Refactor both the UI and the library parts
- Add support for choking and unchoking peers on multiple torrents separately
- And, last but not least, allow for leechers to become seeders while/after leeching.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import React, { useState } from "react";
 import ReactDOM from "react-dom";
 import "./styles.css";
-import WebTorrent, { ClientEvents } from "./webtorrent-custom";
+import WebTorrent from "./webtorrent-custom";
```
Comment (joelalejandro) [inline src/index.js:4]: Can we rename this to `web3torrent-lib` or something like that?

File: src/index.js [hunk 2]
```diff
@@ -13,143 +14,107 @@ const initialState = {
...
-  const [client, setClient] = useState(initialClient);
-  const [status, setStatus] = useState(initialState);
+  let loggerId;
+  const [status, setStatus] = useState(InitialState);
   const [seedMagnet, setSeedMagnet] = useState("");
-  const [allowedPeers, setAllowedPeers] = useState({});
   const [leechMagnet, setLeechMagnet] = useState("");
-
+  const log = progressLogger(loggerId, status, setStatus, setSeedMagnet, seedMagnet);
+  const webClient = client;
```
Comment (joelalejandro) [inline src/index.js]: Why is this necessary?

File: src/index.js (hunk 1 above)
Comment (spersico) [reply src/index.js:4]: Sure! I'll make it in another PR, so we don't lose traceability of the changes done in this PR

File: src/index.js (hunk 2 above)
Comment (spersico) [reply src/index.js]: it isn't
